### DISCUSSION TITLE
How can i get the time of each transaction (Image Attached)

### DISCUSSION THREAD

**Discussion by abeer10** [mempool/mempool#4234] [Q&A]
<img width="631" alt="Screenshot 2023-09-06 at 20 03 00" src="https://github.com/mempool/mempool/assets/17138956/6ba25b93-ddfa-4d1b-98f2-590cc1df2a34">
Last one

**Comment by mononaut** [comment]
Do you mean the time until the next block? (e.g. "In ~10 Minuten")?

If so, that comes from the average block time of the current difficulty epoch, which you can get either:
1. from the difficulty adjustment REST API at `/api/v1/difficulty-adjustment`, which returns a result like:
```json
{
    "progressPercent": 18.154761904761905,
    "difficultyChange": -2.965244635987463,
    "estimatedRetargetDate": 1695217210600,
    "remainingBlocks": 1650,
    "remainingTime": 1023039600,
    "previousRetarget": -2.6452059861325523,
    "previousTime": 1693967242,
    "nextRetargetHeight": 808416,
    "timeAvg": 620024,
    "timeOffset": 0,
    "expectedBlocks": 378.215
}
```
where the `timeAvg` field contains the current average block time in milliseconds.

2. over websocket using the `stats` subscription, which sends the same data in frequent updates in the `da` field of the websocket.
